### ISSUE TITLE
Add new IngressClass resource

### ISSUE THREAD

**Issue by git001** [haproxytech/kubernetes-ingress#171]
In Kubernetes 1.18 is the new IngressClass resource developed.

Doc:  
https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.18.md#extending-ingress-with-and-replacing-a-deprecated-annotation-with-ingressclass  
https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-class

PR: https://github.com/kubernetes/kubernetes/pull/88509

Can the ingress controller be used with the new IngressClass?

**Comment by ailenli** [comment]
this is a  important improvement

**Comment by prometherion** [comment]
@git001 with the following commits we're supporting IngressClass and `ingresses.networking.k8s.io/v1` API! 🎉 

Related commits:
- [BUILD/MINOR: supporting networking.k8s.io ingress and ingressclass](https://github.com/haproxytech/kubernetes-ingress/commit/23c08c3afcdd70da310577a720d12b6324865703)
- [MINOR: ingress: supporting IngressClass by networking.k8s.io](https://github.com/haproxytech/kubernetes-ingress/commit/e6dd2b39cf122238b4f14f9d7215eafb88828c25)
- [MEDIUM: supporting kubernetes 1.19 (networking.k8s.io ingress)](https://github.com/haproxytech/kubernetes-ingress/commit/0cb432a42896ceca8b93a9b724ffce3165be4f74)

Just keep in mind that, starting from 1.19, a dedicated `IngressClass` resource must be created as the following, keeping parity with the Ingress class annotation (or `spec` field) with the `IngressClass` resource name:

```yaml
apiVersion: networking.k8s.io/v1
kind: IngressClass
metadata:
  name: haproxy
spec:
  controller: haproxy.org/ingress-controller
```

We're going to document this upon 1.5 release where this feature is planned to land, along with RBAC updates.

JFI this feature is backward compatible with previous Kubernetes releases, meaning that the IC can be still used with Kubernetes < 1.14

**Comment by Mo3m3n** [comment]
This available from `haproxytech/kubernetes-ingress:1.5.0`
